src/main/java/org/opensearch/ad/transport/ValidateAnomalyDetectorTransportAction.java

```diff
@@ -0,0 +1,273 @@
+                    clusterService,
+                    client,
+                    validateListener,
+                    anomalyDetectionIndices,
+                    detector,
+                    request.getRequestTimeout(),
+                    request.getMaxSingleEntityAnomalyDetectors(),
+                    request.getMaxMultiEntityAnomalyDetectors(),
+                    request.getMaxAnomalyFeatures(),
+                    RestRequest.Method.POST,
+                    xContentRegistry,
+                    user,
+                    searchFeatureDao,
+                    request.getTypeStr()
+                );
+                try {
+                    handler.start();
+                } catch (Exception exception) {
+                    String errorMessage = String
+                        .format(Locale.ROOT, "Unknown exception caught while validating detector %s", request.getDetector());
+                    logger.error(errorMessage, exception);
+                    listener.onFailure(exception);
+                }
+            }
+        }, listener);
+    }
+
+    protected DetectorValidationIssue parseADValidationException(ADValidationException exception) {
+        String originalErrorMessage = exception.getMessage();
+        String errorMessage;
+        Map<String, String> subIssues = null;
+        Object suggestion = null;
+        switch (exception.getType()) {
+            case FEATURE_ATTRIBUTES:
+                int firstLeftBracketIndex = originalErrorMessage.indexOf("[");
+                int lastRightBracketIndex = originalErrorMessage.lastIndexOf("]");
+                if (firstLeftBracketIndex != -1 && lastRightBracketIndex != -1) {
+                    // if feature issue messages are between square brackets like
+                    // [Feature has issue: A, Feature has issue: B]
+                    errorMessage = originalErrorMessage.substring(firstLeftBracketIndex + 1, lastRightBracketIndex);
+                    subIssues = getFeatureSubIssuesFromErrorMessage(errorMessage);
+                } else {
+                    // features having issue like over max feature limit, duplicate feature name, etc.
+                    errorMessage = originalErrorMessage;
+                }
+                break;
+            case NAME:
+            case CATEGORY:
+            case DETECTION_INTERVAL:
+            case FILTER_QUERY:
+            case TIMEFIELD_FIELD:
+            case SHINGLE_SIZE_FIELD:
+            case INDICES:
+                errorMessage = originalErrorMessage;
+                break;
+            default:
+                logger.warn(String.format(Locale.ROOT, "Invalid AD validation exception type: %s", exception));
+                return null;
+        }
+        return new DetectorValidationIssue(exception.getAspect(), exception.getType(), errorMessage, subIssues, suggestion);
+    }
+
+    private Map<String, String> getFeatureSubIssuesFromErrorMessage(String errorMessage) {
```

Comment:
Can you add some comments with an example of a string representation getting parsed into a map here?

Reply:
+ also not sure on where `FEATURE_INVALID_MSG_PREFIX`  would exist in the string, so an example of that would be helpful too